holistics/holistics.py

```diff
@@ -5,95 +5,96 @@
             self.page['_page_size']=_page_size
         if _page is not None:
             self.page['_page']=_page
-        res = self.GetURL(tail_url, self.page)
-        if res!=0:
+        try:
+            res = self.get_url(tail_url, self.page)
             res = res.json()
             while (res['status'] != 'success'):
                 if res['status'] == 'already_existed':
                     self.page['job_id'] = str(res['job_id'])
-                    break
                 if res['status'] == 'failure':
                     print ("Status: Failure")
-                    return 0
-                res = self.GetURL(tail_url, self.page).json()
+                    raise ValueError('Can\'t get export results')
+                res = self.get_url(tail_url, self.page).json()
                 time.sleep(1)
             print ("Success")
             return 1
-        else:
-            return 0
+        except requests.exceptions.HTTPError as err:
+            print ("Fail to get export results")
+            return err
         
-    def DownloadResults(self):
-        print ("Downloading results... ", end='')
-        tail_url = '/exports/download?job_id=' + self.page['job_id']
-        res = self.GetURL(tail_url)
-        if res!=0:
+    def download_results(self, path):
+        path = str(path)
+        try:
+            print ("Downloading results... ", end='')
+            tail_url = '/exports/download?job_id=' + self.page['job_id']        
+            res = self.get_url(tail_url)
             text = str(res.content, 'utf-8', errors='replace')
             data = pd.read_csv(io.StringIO(text))
-            self.data = True
-            print ("Success")
-            if self.path == None:
+            if path == None:
                 return data
             else:
                 try:
-                    data.to_csv(self.path, encoding='utf-8', index=False)
+                    data.to_csv(path, encoding='utf-8', index=False)
                     print ("Export to file successed!")
+                    return data
                 except:
-                    print ("Can't export file to " + self.path)
+                    print ("Can't export file to " + path)
+        except (requests.exceptions.HTTPError,pd.errors.ParserError) as err:
+            print ("Fail to download results")
+            return err
         
 
-    def ExportData(self,report_id,filters=None, page_size=None, page=None):
-        self.SubmitReport(report_id, filters)
-        self.GetExportResults(self.page['job_id'],page_size,page)
-        return self.DownloadResults()
+    def export_data(self,report_id, path=None,filters=None, page_size=None, page=None):
+        try:
+            res = self.submit_export(report_id, filters)
+            res = self.get_export_results(self.page['job_id'],page_size,page)
+            res = self.download_results(path)
+            if isinstance(res, ((requests.exceptions.HTTPError, pd.errors.ParserError, ValueError))):
```

Comment:
- Do not handle errors by checking res like this. In the inner functions, use `raise` instead of `return` the error
- In which case will res be a ValueError?

Fixing the first bullet point is enough